src/name_bazaar/ui/pages/manage_names_page.cljs

```diff
@@ -0,0 +1,256 @@
+(ns name-bazaar.ui.pages.manage-names-page
+  (:require
+    [cljs-time.coerce :as time-coerce :refer [to-epoch to-date from-date]]
+    [cljs-time.core :as t]
+    [district0x.shared.utils :refer [pos-ether-value? empty-address?]]
+    [district0x.ui.components.input :refer [input token-input]]
+    [district0x.ui.components.misc :refer [page]]
+    [district0x.ui.components.transaction-button :refer [transaction-button]]
+    [district0x.ui.utils :refer [date+time->local-date-time format-time-duration-units]]
+    [name-bazaar.shared.utils :refer [top-level-name?]]
+    [name-bazaar.ui.components.app-layout :refer [app-layout]]
+    [name-bazaar.ui.components.date-picker :refer [date-picker]]
+    [name-bazaar.ui.components.ens-record.ens-name-input :refer [ens-name-input]]
+    [name-bazaar.ui.components.loading-placeholders :refer [content-placeholder]]
+    [name-bazaar.ui.components.offering.offering-type-select :refer [offering-type-select]]
+    [name-bazaar.ui.constants :as constants]
+    [name-bazaar.ui.utils :refer [reverse-record-node namehash sha3 normalize strip-root-registrar-suffix valid-ens-name? path-for]]
+    [re-frame.core :refer [subscribe dispatch]]
+    [reagent.core :as r]
+    [cljs-web3.core :as web3]
+    [soda-ash.core :as ui]))
+
+(def ownership-status->text
+  {:ens.ownership-status/empty-name ""
+   :ens.ownership-status/loading "Checking ownership..."
+   :ens.ownership-status/not-ens-record-owner "You are not owner of this name"
+   :ens.ownership-status/not-deed-owner "You don't own this name's locked value"
+   :ens.ownership-status/owner "You are owner of this name"})
+
+(defn default-point-name-form-data [addr]
+  {:ens.record/address (or addr "0x")
+   :ens.record/name ""}
+  )
+
+(defn load-name-ownership [value]
+  (let [full-name (str value constants/registrar-root)
+        node (namehash full-name)]
+    (dispatch [:ens.records/load [node] {:load-resolver? true}])
+    (when (top-level-name? full-name)
+      (dispatch [:registrar.entries/load [(sha3 value)]]))))
+
+(defn load-resolver [address]
+  (let [node (reverse-record-node address)]
+    (dispatch [:ens.records.resolver/load [node]])))
+
+(defn ens-name-text-field [{:keys [:value]}]
```

Comment:
This component should be abstracted away and the same code should be used for `name-bazaar.ui.pages.offering-create-page`, not duplicated